### PR TITLE
Port `android/`

### PR DESCRIPTION
**Pull request by scarf005**
#### Summary

SUMMARY: Build "Port DDA's more recent version of android/"

#### Purpose of change

- very outdated
- fails on #2341
- may improve #2413 

#### Describe the solution

~~blocked by~~ ~~is~~ was blocking #2501
- port the entire `android/` directory from DDA
- port https://github.com/CleverRaven/Cataclysm-DDA/pull/52633
- https://github.com/CleverRaven/Cataclysm-DDA/pull/48371
- migrate to androidx, idk what it does but intellij idea says it's good for API stability
- bump JDK version to 17

#### Describe alternatives you've considered

- set up specific android SDK version on GA.
- migrating the build script from groovy to kotlin. it's a giant, potentially breaking change so it'd better be a separate PR.
- change namespace from dda to bn

#### Testing

successfully built and installed on android emulator on intellij idea with android plugin.

### CODE REVIEW

**Comment by scarf005** [comment]
curprit was a funni typo in `androidmanifest.xml`

**Comment by scarf005** [comment]
image
*finally*

File: android/app/src/main/java/com/cataclysmbnteam/cataclysmbn/NativeUI.java
Comment (olanti-p) [inline android/app/src/main/java/com/cataclysmbnteam/cataclysmbn/NativeUI.java:1]: This file looks like a part of https://github.com/CleverRaven/Cataclysm-DDA/pull/53132, but we don't have that functionality in BN.

File: android/app/jni/src/third-party/Android.mk
Comment (olanti-p) [inline android/app/jni/src/third-party/Android.mk:1]: We don't have `src/third-party/`, nor the flatbuffers support mentioned in this file.

I don't think this file is relevant for BN.

File: android/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -4,4 +4,22 @@
     <string name="upgradeTitle">Upgrading game data...</string>
     <string name="helpTitle">Help &amp; Controls</string>
     <string name="helpMessage">"Swipe" for directional movement (hold for virtual joystick). "Tap" to confirm selection in menu or Pause one turn in-game (hold to Pause several turns in-game). "Double tap" to cancel or go back in menus (works like Escape key). "Pinch" to zoom in/out (in-game). Use hardware "Back" button to toggle virtual keyboard (hold to toggle keyboard shortcuts).</string>
+    <string name="accessibilityServicesTitle">Accessibility Services Warning</string>
+    <string name="accessibilityServicesMessage">If swipe commands and shortcuts menus are not working in game, you might want to disable one of the following accessibility services/software that interact with touchscreen (swipe assists, auto-clickers, one-handed mode, etc.):\n%1$s</string>
+    <string name="softwareRendering">Software rendering</string>
+    <string name="forceFullscreen">Force fullscreen</string>
+    <string name="trapBackButton">Trap Back button</string>
+    <string name="nativeAndroidUI">Native Android UI menus</string>
```
Comment (olanti-p) [inline android/app/src/main/res/values/strings.xml:12]: It's mentioning the "Native UI" we don't have, and I assume "Yes/No/OK/Cancel" may also be part of that.

**Review by olanti-p — COMMENTED** [review]
There's a lot of bits that are relevant to DDA-specific functionality, but I think it would be less hassle to leave them in and let them rot rather than trying to clean everything out and ending up with billion merge conflicts in the future.

I also don't understand the meaning behind most of the changes in this PR, but if it doesn't break builds I think it would be fine to merge it as is. It's not like we haven't been flying mostly blind with Android builds for the entire time.

**Comment by olanti-p** [comment]
I tried rebasing it on latest `upload` and building it with Github Actions, and it seems to work.
https://github.com/olanti-p/Cataclysm-BN/releases/tag/cbn-experimental-2023-04-06-2101
Running it, however, results in the same crash as described in #2413
Latest BN release, [2023-04-05-1022](https://github.com/cataclysmbnteam/Cataclysm-BN/releases/tag/cbn-experimental-2023-04-05-1022), works fine.

Android 11

**Comment by scarf005** [comment]
closing as #3572 uses other solution for android

**Pull request closed without merging**